File: c#/Asset/Folders/UpdateFolderMetadata.cs
/*
   Marketo REST API Sample Code
   Copyright (C) 2016 Marketo, Inc.

   This software may be modified and distributed under the terms
   of the MIT license.  See the LICENSE file for details.
*/
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Net;
using System.IO;
using System.Collections;
using Newtonsoft.Json;
using System.Web;

namespace Samples
{

    class UpdateFolder
    {
        private String host = "CHANGE ME"; //host of your marketo instance, https://AAA-BBB-CCC.mktorest.com
        private String clientId = "CHANGE ME"; //clientId from admin > Launchpoint
        private String clientSecret = "CHANGE ME"; //clientSecret from admin > Launchpoint
        public int id;//id of folder to update, required
        public String type;//type of folder, Folder or Program, required
        public String description;//description of folder to update
        public String name;//name of folder to update
        public Boolean isArchive;//option to archive target folder

        /*
        public static void Main(string[] args)
        {
            var folder = new UpdateFolder();
            folder.id = 5565;
            folder.description = "Folder updated by C# example";
            String result = folder.postData();
            Console.WriteLine(result);
        }
        */

        public String postData()
        {
            //Assemble the URL
            String url = host + "/rest/asset/v1/folder/" + id + ".json?access_token=" + getToken();
            //Serialize the Data
            String requestBody = bodyBuilder();
            HttpWebRequest request = (HttpWebRequest)WebRequest.Create(url);
            request.Method = "POST";
            request.ContentType = "application/x-www-form-urlencoded";
            request.Accept = "application/json";
            StreamWriter wr = new StreamWriter(request.GetRequestStream());
            wr.Write(requestBody);
            wr.Flush();
            HttpWebResponse response = (HttpWebResponse)request.GetResponse();
            Stream resStream = response.GetResponseStream();
            StreamReader reader = new StreamReader(resStream);
            return reader.ReadToEnd();
        }
        private String bodyBuilder()
        {
            var body = HttpUtility.ParseQueryString(string.Empty);
            body.Add("type", type);
            if (description != null)
            {
                body.Add("description", description);
            }
            if (name != null)
            {
                body.Add("name", name);
            }
            if (isArchive != null)
            {
                body.Add("isArchive", isArchive.ToString());
            }
            return body.ToString();
        }

        private String getToken()
        {
            String url = host + "/identity/oauth/token?grant_type=client_credentials&client_id=" + clientId + "&client_secret=" + clientSecret;
            HttpWebRequest request = (HttpWebRequest)WebRequest.Create(url);
            request.ContentType = "application/json";
            HttpWebResponse response = (HttpWebResponse)request.GetResponse();
            Stream resStream = response.GetResponseStream();
            StreamReader reader = new StreamReader(resStream);
            String json = reader.ReadToEnd();
            //Dictionary<String, Object> dict = JavaScriptSerializer.DeserializeObject(reader.ReadToEnd);
            Dictionary<String, String> dict = JsonConvert.DeserializeObject<Dictionary<String, String>>(json);
            return dict["access_token"];
        }
    }
}

File: php/Asset/Folders/UpdateFolderMetadata.php
<?php
/*
   UpdateFolderMetadata.php

   Marketo REST API Sample Code
   Copyright (C) 2016 Marketo, Inc.

   This software may be modified and distributed under the terms
   of the MIT license.  See the LICENSE file for details.
*/
$folder = new UpdateFolder();
$folder->type = "Folder";
$folder->id = 5566;
$folder->description = "This is a folder that has been updated";
print_r($folder->postData());

class UpdateFolder{
	private $host = "CHANGE ME";
	private $clientId = "CHANGE ME";
	private $clientSecret = "CHANGE ME";
	public $id;//id of folder to update, required
	public $type;//type of folder, Folder or Program, required
	public $name;//name of folder to create
	public $parent;//JSON object representing a folder with two members, id and type(Folder or Program), required
	public $description;//optional description for new folder
	public $isArchive; //boolean, set to true to archive folder
	
	public function postData(){
		$url = $this->host . "/rest/asset/v1/folder/" . $this->id . ".json?access_token=" . $this->getToken() . "&type=" . $this->type;
		$ch = curl_init($url);
		$requestBody = $this->bodyBuilder();
		curl_setopt($ch,  CURLOPT_RETURNTRANSFER, 1);
		curl_setopt($ch, CURLOPT_HTTPHEADER, array('accept: application/json'));
		curl_setopt($ch, CURLOPT_POST, 1);
		curl_setopt($ch, CURLOPT_POSTFIELDS, $requestBody);
		curl_getinfo($ch);
		$response = curl_exec($ch);
		return $response;
	}
	
	private function getToken(){
		$ch = curl_init($this->host . "/identity/oauth/token?grant_type=client_credentials&client_id=" . $this->clientId . "&client_secret=" . $this->clientSecret);
		curl_setopt($ch,  CURLOPT_RETURNTRANSFER, 1);
		curl_setopt($ch, CURLOPT_HTTPHEADER, array('accept: application/json',));
		$response = json_decode(curl_exec($ch));
		curl_close($ch);
		$token = $response->access_token;
		return $token;
	}
	private function bodyBuilder(){
		$requestBody = "";
		if (isset($this->name)){
			$requestBody .= "name=" . $this->name;
		}
		if (isset($this->description)){
			$requestBody .= "&description=" . $this->description;
		}
		if (isset($this->isArchive)){
			$requestBody .= "&isArchive=" . $this->isArchive;
		}
		return $requestBody;
	}	
}